plugins/inputs/nsd/README.md

```diff
@@ -0,0 +1,173 @@
+# NSD Input Plugin
+
+This plugin gathers stats from [NSD](https://nlnetlabs.nl/projects/nsd/about/) - an authoritative DNS server.
+
+### Configuration
+
+```toml
+ # A plugin to collect stats from the NSD DNS server
+ [[inputs.nsd]]
+   ## Address of server to connect to, read from nsd conf default, optionally ':port'
+   ## Will lookup IP if given a hostname
+   server = "127.0.0.1:8952"
+ 
+   ## If running as a restricted user you can prepend sudo for additional access:
+   # use_sudo = false
+ 
+   ## The default location of the nsd-control binary can be overridden with:
+   # binary = "/usr/sbin/nsd-control"
+ 
+   ## The default timeout of 1s can be overriden with:
+   # timeout = "1s"
+ 
+   ## When set to true, thread metrics are tagged with the thread id.
+   ##
+   ## The default is false for backwards compatibility, and will be change to
+   ## true in a future version.  It is recommended to set to true on new
+   ## deployments.
+   thread_as_tag = false
```

Comment:
In nsd the only per thread metric available is the number of queries, which has the prefix serverX.
So i wasn't sure if this should be a tag by default or that we should give a choice. But now that i looked through the unbound commit history i saw that it was only because of legacy versions of the plugin and probably should be true by default here.

Reply:
Is this the `server0.queries=32` option?  Yeah, we should make this true by default, no need for an option.  If it is this server option then we could have it be `nsd,server=0 queries=32i`

Reply:
i made it default, but i need a separate measurement for it "nsd_server". Or is there any other way i can pass the test hasTags() with 2 measurements with the same name but one with tag and the other without?

Reply:
Probably the easiest way is to get the `acc.Metrics`, check which one has the tag, and then compare them against an expected metric created with `testutil.MustMetric` using `testutil.RequireMetricEqual`.

Reply:
I made a check to find the metric with the tag (both have the same name now)